sysgen/sysgen.go

```diff
@@ -325,6 +325,28 @@ func generateStructs(desc *Description, consts map[string]uint64, out io.Writer)
 	fmt.Fprintf(out, "}\n")
 }
 
+func parseRange(buffer string, consts map[string]uint64) (string, string) {
+	var lookupConst func(name string) string
+	lookupConst = func(name string) string {
```

Comment:
lookupConst := ...
and drop the previous line

You may have seen this form with separate declaration and initialization in sys/prog, but it is necessary only when you have recursion (lookupConst would need to call lookupConst). Otherwise you just do `f := func(....`.